VTUL/iawa
# Hyrax 2.1.0 IAWA merge into master

**JIRA Tickets**: 
* https://webapps.es.vt.edu/jira/browse/LIBTD-1269
* https://webapps.es.vt.edu/jira/browse/LIBTD-1270
* https://webapps.es.vt.edu/jira/browse/LIBTD-1271
* https://webapps.es.vt.edu/jira/browse/LIBTD-1272
* https://webapps.es.vt.edu/jira/browse/LIBTD-1273
* https://webapps.es.vt.edu/jira/browse/LIBTD-1257

Related Links: This PR's merge should be coordinated with https://github.com/VTUL/InstallScripts/pull/175

# What does this Pull Request do?
This PR merges to master the main work for migrating IAWA to Hyrax 2.1.0.

# What are the changes?
There were multiple changes related to the migration of the IAWA application from Hyrax 1.0.4 to Hyrax 2.1. Some of them can be found in the migration notes:
* https://github.com/VTUL/iawa/wiki/IAWA-Migration-Notes:-Hyrax-v.-2.0.0-to-v.-2.1.0
* https://github.com/VTUL/iawa/wiki/IAWA-Migration-Notes%3A-Hyrax-v1.0.4-to-v.2.0.0

# How should this be tested?

To test this, please use the InstallScripts branch `LIBTD-1257`. This will ensure that the appropriate default Hyrax CollectionTypes are installed. Also, use the following settings in your `site_secrets.yml` :

```
project_name: 'iawa'
project_git_identifier: 'LIBTD-1256'
project_app_env: 'production'
```

Note that the `project_secret_key_base` and `project_cas_url` will also need to be updated.

Try the following:
* Attempt to create a collection, both with and without branding images.
* Attempt to create one or more works, both with and without images. 
* Attempt to associate a work with a collection.
* Attempt to modify your user profile avatar.
* Attempt to access the dashboard from the upper right drop down.
* Try anything else and note any irregularities. 

# Additional Notes:
* This PR does not address any existing import scripts from the Hyrax 1.0 version, which may be affected with this change when it is eventually merged into master.
* This PR needs to be tested in production mode due to an open issue with the development mode and the Mirador viewer: https://webapps.es.vt.edu/jira/browse/LIBTD-1343

# Interested parties
@tingtingjh
